src/back/glsl/mod.rs

```diff
@@ -3146,10 +3146,16 @@ impl<'a, W: Write> Writer<'a, W> {
             None => {
                 if tex_1d_hack {
                     write!(self.out, "ivec2(")?;
+                } else if vector_size == 1 {
+                    write!(self.out, "int(")?;
+                } else {
+                    write!(self.out, "ivec{}(", vector_size)?;
```

Comment:
Let's do this only if necessary (i.e. only if the `coordinate` expression's type is unsigned).